src/core/opamStd.ml

```diff
@@ -920,50 +931,119 @@ module OpamSys = struct
+  let guess_shell_compat () =
+    let parent_guess () =
+      let ppid = Unix.getppid () in
+      let dir = Filename.concat "/proc" (string_of_int ppid) in
+      try
+        Some (Unix.readlink (Filename.concat dir "exe"))
+      with e ->
+        fatal e;
         try
-          Some (Unix.readlink (Filename.concat dir "exe"))
-        with e ->
-          fatal e;
-          try
-            with_process_in "ps"
-              (Printf.sprintf "-p %d -o comm= 2>/dev/null" ppid)
-              (fun ic -> Some (input_line ic))
-          with
-          | Unix.Unix_error _ | Sys_error _ | Failure _ | End_of_file | Not_found ->
-              try
-                let c = open_in_bin ("/proc/" ^ string_of_int ppid ^ "/cmdline") in
-                begin try
-                  let s = input_line c in
+          with_process_in "ps"
+            (Printf.sprintf "-p %d -o comm= 2>/dev/null" ppid)
+            (fun ic -> Some (input_line ic))
+        with
+        | Unix.Unix_error _ | Sys_error _ | Failure _ | End_of_file | Not_found ->
+            try
+              let c = open_in_bin ("/proc/" ^ string_of_int ppid ^ "/cmdline") in
+              begin try
+                let s = input_line c in
+                close_in c;
+                Some (String.sub s 0 (String.index s '\000'))
+              with
+              | Not_found ->
+                  None
+              | e ->
                   close_in c;
-                  Some (String.sub s 0 (String.index s '\000'))
-                with
-                | Not_found ->
-                    None
-                | e ->
-                    close_in c;
-                    fatal e; None
-                end
-              with e ->
-                fatal e; None
+                  fatal e; None
+              end
+            with e ->
+              fatal e; None
     in
     let test shell = shell_of_string (Filename.basename shell) in
-    let shell =
-      match Option.replace test parent_guess with
-      | None ->
+    if Sys.win32 then
+      let shell =
+        match Lazy.force windows_get_shell with
+        | None ->
           Option.of_Not_found Env.get "SHELL" |> Option.replace test
-      | some ->
+        | some ->
           some
-    in
-    Option.default SH_sh shell
+        in
+      Option.default windows_default_shell shell
+    else
+      let shell =
+        match Option.replace test (parent_guess ()) with
+        | None ->
+            Option.of_Not_found Env.get "SHELL" |> Option.replace test
+        | some ->
+            some
+      in
+      Option.default unix_default_shell shell
 
   let guess_dot_profile shell =
+    let win_my_powershell f =
+      let p = Filename.concat (home ()) "Documents" in
+      if Sys.file_exists p then Filename.concat (Filename.concat p "PowerShell") f
+      else let p = Filename.concat (home ()) "My Documents" in
+      if Sys.file_exists p then Filename.concat (Filename.concat p "PowerShell") f
+      else f
+    in
```

Comment:
The PowerShell documentation is really badly written - PowerShell actually reads FOLDERID_Documents (which then takes folder redirection into account), but in fact I have another idea to query PowerShell for it instead, but that can wait for a subsequent PR!

Reply:
Ok